cloud-functions/upload_fhir_bundle/main.py

```diff
@@ -82,24 +92,60 @@ def upload_fhir_bundle(request: flask.Request) -> flask.Response:
     ]
     fhir_store_url = "/".join(fhir_store_url)
 
-    # Upload bundle to the FHIR store using the GCP Crednetial Manager for
+    # Upload bundle to the FHIR store using the GCP Credential Manager for
     # authentication.
-    response = upload_bundle_to_fhir_server(
+    fhir_store_response = upload_bundle_to_fhir_server(
         request_body.bundle, credential_manager, fhir_store_url
     )
+    fhir_store_response_body = fhir_store_response.json()
     # the response from PHDI is a request.Response which
     #   then needs to be translated into a flask.Response object
-    #
-    # If the response is an error then grab the data and store as a message
-    # otherwise get the json and store in the json_payload of the
-    # flask response
-    if response.status_code == 200:
-        final_response = make_response(
-            status_code=response.status_code, json_payload=response.json()
+
+    # If the FHIR store responds with a 200 check if any individual resources failed to
+    # upload.
+    failed_resources = []
+    if fhir_store_response.status_code == 200:
+        failed_resources = [
+            entry
+            for entry in fhir_store_response_body["entry"]
+            if entry["response"]["status"] not in ["200 OK", "201 Created"]
+        ]
+
+        fhir_store_response_body = {
+            "entry": failed_resources,
+            "resourceType": "Bundle",
+            "type": "transaction-response",
+        }
```

Comment:
I'm assuming that this is just adding the resourceType and type for each of the different failures, correct?

Reply:
It will be whatever information the FHIR server returns about the upload failure of individual resources.